presto-hive/src/main/java/com/facebook/presto/hive/RecordFileWriter.java

```diff
@@ -161,6 +163,8 @@ public void rollback()
     {
         try {
             recordWriter.close(true);
+            // perform explicit delete of written file here as recordWriter.close() ignore the abort flag.
+            path.getFileSystem(conf).delete(path, false);
```

Comment:
What will happen if the created file is not yet visible? I wasn't able to find an answer to this question here:

http://docs.aws.amazon.com/AmazonS3/latest/dev/Introduction.html#ConsistencyModel

but it states that: `A process writes a new object to Amazon S3 and immediately lists keys within its bucket. Until the change is fully propagated, the object might not appear in the list. ` However there is no explanation what happens with "delete-after-write" :/